extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreRecorder.java

```diff
@@ -259,9 +259,13 @@ private static VertxOptions convertToVertxOptions(VertxConfiguration conf, Vertx
                 if (!tmp.mkdirs()) {
                     LOGGER.warnf("Unable to create Vert.x cache directory : %s", tmp.getAbsolutePath());
                 }
-                if (!(tmp.setReadable(true, false) && tmp.setWritable(true, false))) {
-                    LOGGER.warnf("Unable to make the Vert.x cache directory (%s) world readable and writable",
-                            tmp.getAbsolutePath());
+                String os = System.getProperty("os.name").toLowerCase();
+                if (!os.contains("win")) {
```

Comment:
Are you sure macOS won't be flagged (Darwin)?

Probably better to have a look how it's done in Smallrye Common here: https://github.com/smallrye/smallrye-common/blob/main/os/src/main/java/io/smallrye/common/os/OS.java#L46 .

Reply:
Fixed, using the same approach.